Regression test fixes

SparkiDev opened the pull request:
./configure --enable-all --disable-rsa
./configure --disable-chacha --disable-asm
./configure --disable-rsa --disable-ecc --disable-dh --enable-curve25519
./configure --enable-cryptonly (and ed25519, curve448, ed448)
./configure --disable-tls13 --enable-psk --disable-rsa --disable-ecc
./configure --disable-dh C_EXTRA_FLAGS=-DWOLFSSL_STATIC_PSK
./configure --disable-oldtls --enable-psk -disable-rsa --disable-dh
./configure --disable-ecc --disable-asn C_EXTRA_FLAGS=-DWOLFSSL_STATIC_PSK
./configure --enable-lowresource --enable-singlethreaded --disable-asm
./configure --disable-errorstrings --disable-pkcs12 --disable-sha3 --disable-sha224
./configure --disable-sha384 --disable-sha512 --disable-sha --disable-md5
./configure --disable-aescbc --disable-chacha --disable-poly1305 --disable-coding
Various build combinations with WOLFSSL_SP_MATH and WOLFSSL_SP_MATH_ALL

dgarske reviewed (changes requested):
Awesome regression. 

Still seeing build errors with:
```
./configure --disable-rsa --disable-ecc --disable-dh --enable-curve25519 && make
./configure --disable-aescbc --disable-chacha --disable-poly1305 --disable-coding && make
```

SparkiDev commented:
./configure --disable-rsa --disable-ecc --disable-dh --enable-curve25519 && make
Can't build as no cipher suites. Fixed build errors and put in a check and report an error.
./configure --disable-aescbc --disable-chacha --disable-poly1305 --disable-coding && make
Won't pass tests as our tests use PEM files. Fixed to compile.
